cmd/libsnap-confine-private/string-utils.c

```diff
@@ -69,3 +70,29 @@ int sc_must_snprintf(char *str, size_t size, const char *format, ...)
 
 	return n;
 }
+
+void sc_string_append(char *buf, size_t buf_size, const char *str)
```

Comment:
Can you rename these as:
```
size_t sc_string_append(char *dst, size_t dst_size, const char *str)
```

This will make things more readable IMHO and lets the caller know the new size of dst (buf).